src/invidious/yt_backend/connection_pool.cr

```diff
@@ -26,17 +26,23 @@ struct YoutubeConnectionPool
   end
 
   def client(region = nil, &block)
-    if region
+    if !CONFIG.http_proxy && region
       conn = make_client(url, region)
+      conn.proxy = make_configured_http_proxy_client() if CONFIG.http_proxy
```

Comment:
This will never run because of line 29

Reply:
Good catch. The proxy decision should probably be fully independent from the region decision.